zio-http/src/main/scala/zio/http/security/OutputEncoder.scala

```diff
@@ -0,0 +1,35 @@
+package zio.http.security
+
+private[http] object OutputEncoder {
+
+  /**
+   * Encode HTML characters that can cause XSS, according to OWASP
+   * specification:
+   * https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html#output-encoding-rules-summary
+   *
+   * Specification: Convert & to &amp;, Convert < to &lt;, Convert > to &gt;,
+   * Convert " to &quot;, Convert ' to &#x27;, Convert / to &#x2F;
+   *
+   * Only use this function to encode characters inside HTML context:
+   * <html>output</html
+   *
+   * @param output
+   *   string that needs HTML encoding
+   * @return
+   *   HTML encoded string
+   */
+  def encodeHtml(output: String): String = {
+    output.map(char => encodeHtmlChar(char))
+  }.mkString
+
+  private def encodeHtmlChar(char: Char): String = char match {
+    case '&'     => "&amp"
+    case '<'     => "&lt;"
+    case '>'     => "&gt;"
+    case '"'     => "&quot;"
+    case '\''    => "&#x27;"
+    case '/'     => "&#x2F;"
+    case _ @char => char.toString
+  }
```

Comment:
We can optimize the performance a bit here —

1. Initialize with `StringBuilder` with the expected length of the output string.
2. Save the output string allocations as a Charsequence inside of `OutputEncoder` once.

Reply:
Thx for the review, changes have been made. Not sure if your 2. is met, could you have a look?

Reply:
If you save the encoded versions of each char in a val you won't be allocating it for every call.

Reply:
Got it, pushed the changes.